external/fv3fit/fv3fit/emulation/thermobasis/emulator.py

```diff
@@ -0,0 +1,366 @@
+        group.add_argument("--v-weight", default=100.0, type=float)
+        group.add_argument("--t-weight", default=100.0, type=float)
+        group.add_argument("--levels", default="", type=str)
+        group.add_argument(
+            "--no-weight-sharing",
+            dest="weight_sharing",
+            action="store_true",
+            help="Weights not shared by any inputs.",
+        )
+
+        group = parser.add_argument_group("single level output")
+        group.add_argument(
+            "--relative-humidity",
+            action="store_true",
+            help="if true use relative based prediction.",
+        )
+        group.add_argument(
+            "--cloud-water",
+            action="store_true",
+            help="if true predict the cloud water field.",
+        )
+        group.add_argument("--variable", default=3, type=int)
+        group.add_argument("--scale", default=1.0, type=float)
+
+        group = parser.add_argument_group("network structure")
+        group.add_argument("--num-hidden", default=256, type=int)
+        group.add_argument("--num-hidden-layers", default=3, type=int)
+        group.add_argument(
+            "--extra-variables",
+            default="",
+            type=str,
+            help="comma separated list of variable names.",
+        )
+
+    @staticmethod
+    def from_args(args) -> "Config":
+        config = Config()
+        config.batch_size = args.batch_size
+        config.epochs = args.epochs
+        config.learning_rate = args.lr
+        config.momentum = args.momentum
+        config.batch = BatchDataConfig(args.training_data, args.testing_data)
+
+        config.num_hidden = args.num_hidden
+        config.num_hidden_layers = args.num_hidden_layers
+        config.wandb_logger = args.wandb
+        config.relative_humidity = args.relative_humidity
+        config.cloud_water = args.cloud_water
+
+        if args.level:
+            if args.relative_humidity:
+                config.target = RHLoss(level=args.level, scale=args.scale)
+            else:
+                config.target = QVLoss(args.level, scale=args.scale)
+        elif args.multi_output:
+            levels = [int(s) for s in args.levels.split(",") if s]
+            config.target = MultiVariableLoss(
+                levels=levels,
+                qc_weight=args.qc_weight,
+                q_weight=args.q_weight,
+                u_weight=args.u_weight,
+                v_weight=args.v_weight,
+                t_weight=args.t_weight,
+                rh_weight=args.rh_weight,
+            )
+            config.weight_sharing = args.weight_sharing
+        else:
+            raise NotImplementedError(
+                f"No problem type detected. "
+                "Need to pass either --level or --multi-output"
+            )
+
+        if args.extra_variables:
+            config.extra_input_variables = args.extra_variables.split(",")
+
+        return config
+
+
+class Emulator:
```

Comment:
This doesn't look like a generic class I should subclass from if I want to make an emulation model, and nothing about it is really specific to emulation. Could you make it clear in the documentation and naming what the purpose of this is, and how a new user to this part of the code would consider making use of or extending this code (if at all)?

Reply:
I agree this would be nice, but I'm okay violating the open-closed rule when it comes to these kinds of simple factories functions. If we start to heavily extend this model, I agree we will need a better plugin architecture. As long as this function fits on one screen, I would advocate for leaving it unchanged.

Reply:
see 41c46d34baf5f15de1cb6eefc2eb8de33e806279